File: short_problems/C++/is_fibo1.cpp
#include <iostream>

using namespace std;

bool is_fibo(unsigned long long num)
{
    if (num == 0 || num == 1) {
        return true;
    }
    unsigned long long prev = 0;
    unsigned long long crt = 1;
    while (crt < num) {
        unsigned long long tmp = crt;
        crt += prev;
        prev = tmp;
    }

    return num == crt;
}

int main()
{
    int num_cases;
    cin >> num_cases;
    for (int i = 0; i < num_cases; i++) {
        unsigned long long num;
        cin >> num;
        if (is_fibo(num)) {
            cout << "IsFibo" << endl;
        } else {
            cout << "IsNotFibo" << endl;
        }
    }
    return 0;
}

File: short_problems/Python/is_fibo1.py
#!/usr/local/bin/python3

def is_fibo(num):
    if num == 0 or num == 1:
        return True
    prev, crt = 0, 1
    while crt < num:
        tmp = crt
        crt = crt + prev
        prev = tmp

    return num == crt

if __name__ == "__main__":
    num_cases = int(input())
    for i in range(num_cases):
        num = int(input())
        if is_fibo(num):
            print("IsFibo")
        else:
            print("IsNotFibo")
